src/Service/IndexConfiguration.php

```diff
@@ -35,6 +35,8 @@ class IndexConfiguration
 
     private static bool $auto_dependency_tracking = true;
 
+    private static bool $use_synchronous_dependencies = true;
```

Comment:
I think this is dataobject specific so should live on DataObjectDocument rather than index config